### ISSUE TITLE
Munge script modifies the Z score values, may be a bug??

### ISSUE THREAD

**Issue by drveera** [bulik/ldsc#98]
Hi,

When I provide a summary statistics with Z scores as input to munge script, it outputs completely different Z scores in the munged file, which makes no sense. All the stats are changing because of this.  I also tried munging a sumstats with SE and beta and  used the output with Z score as input again to munge script and now it complains that the median value of Z score doesn't make sense. Something is definitely wrong with munge script the way it handles the Z score column. This might be a serious issue. Could you please verify and let us know what's the problem. 

Regards
Veera Manikandan Rajagopal MD
PhD Student
Department of Biomedicine
Aarhus University
Aarhus, Denmark

**Comment by rkwalters** [comment]
Hi Veera,
Is this the same issue as your email to the user group about the imputed
summary statistics or something additional?

My initial guess would some issue with parsing the input file's columns
leading it to grab something other than the Z scores, but hard to debug
without more info. Could you send a copy of the log from munge_sumstats.py
and an example of the format of your input file? (e.g. just the first
couple rows, with dummy numeric values if you prefer)

Cheers,
Raymond

**Comment by drveera** [comment]
Hi Raymond,

Its just any summary statistics with just Z score without beta and se. For example can you check with the Intelligence sumstats available from CTG lab website (https://ctg.cncr.nl/software/summary_statistics). After munging the sumstats, please compare the Z score in the munged sumstats with the original one. I tried with the latest version of munge script from the github. 

Regards
Veera

**Comment by rkwalters** [comment]
Hi Veera,
I'm so far unable to recreate this error with those intelligence results.
The Z scores from munge match the input file to within .001 rounding error.
Again, more detail on your munge input/output would be helpful?
Cheers,
Raymond


On Jan 9, 2018 1:07 PM, "Veera" <notifications@github.com> wrote:

Hi Raymond,

Its just any summary statistics with just Z score without beta and se. For
example can you check with the Intelligence sumstats available from CTG lab
website (https://ctg.cncr.nl/software/summary_statistics). After munging
the sumstats, please compare the Z score in the munged sumstats with the
original one. I tried with the latest version of munge script from the
github.

Regards
Veera

—
You are receiving this because you commented.

Reply to this email directly, view it on GitHub
<https://github.com/bulik/ldsc/issues/98#issuecomment-356366091>, or mute
the thread

**Comment by drveera** [comment]
Hi Raymond,

Thanks a lot for trying. I am very sorry, the bug was actually due to modifications in the script made by Jonatan from our lab. We found it now.  I didn't realise in the first place, that I was pulling the changes from his forked version. 

Regards
Veera

**Comment by drveera** [comment]
Hi Raymond, 

Sorry for the confusion. 
still the munge script changes the Z score for my sumstats atleast. Can you please check with the sample file I attached in this comment ?

[sample.sumstats.txt](https://github.com/bulik/ldsc/files/1619349/sample.sumstats.txt)

**Comment by drveera** [comment]
Hi Raymond, 

Digging in to the code I found that there is a function called `p_to_z` which converts P values to Z scores using the formula `np.sqrt(chi2.isf(P,1))`. It seems always does that no matter you provide Z score in the input data. For example , if I have Z score 1.473 and the corresponding P value is 0.07, but the Z score calculated using p_to_z in the script is 1.81. This way it inflates the Z scores. 
<img width="755" alt="screen shot 2018-01-11 at 12 26 59" src="https://user-images.githubusercontent.com/8498890/34823496-dc19d4a8-f6ca-11e7-92dd-c7d424fe1e1f.png">

here in the plot you can see that the Z scores from munge output is inflated compared to original ones. Because of this my GC lambda shoots to 2.7 compared to 1.02 (calculated from original Z scores). 
Kindly help me with this issue. Because LDhub, MTAG all this uses this munge script and so this is an issue at all analysis. 

Thanks

Regards
Veera

**Comment by drveera** [comment]
Hi Raymond,

Little more digging, I found my problem. I calculated the P values from Z scores for the imputed statistics using the formula `pnorm(abs(Z),lower.tail=FALSE)` in R. I missed to multiply the values by 2 and hence I got wrong P values, which I used along with z scores for munging. Hence the resulting conversion of these wrong P values to Z scores during munging amplified the Z scores. Its a very silly mistake I did but I would have missed it if its not for munge script. Though i was scratching my head for the past two days, it was worth it. At the end of the day munge script saved me, also I have more understanding of the P values and Z scores relationship. Everything is fine now 👍 

Regards
Veera
